engine/collection/ingest/engine.go

```diff
@@ -355,8 +357,8 @@ func (e *Engine) ingestTransaction(
 		return nil
 	}
 
-	// check if the transaction is valid
-	err = e.transactionValidator.Validate(tx)
+	// we don't pass actual ctx as we don't execute any scripts inside for now
+	err = e.transactionValidator.Validate(context.Background(), tx)
```

Comment:
since you're using a background context here, you can revert all of the other context changes in this file since they're not needed anymore